CMakeLists.txt

```diff
@@ -76,9 +76,28 @@ set(XL_Fortran_FLAGS_DEBUG -O0 -g)
 set(XL_Fortran_FLAGS_RELEASE -O3 -g)
 set(XL_Fortran_FLAGS_RELWITHDEBINFO -O3 -g -DNDEBUG)
 
-set(Clang_C_FLAGS_DEBUG -O0 -g -save-temps -std=c99)
-set(Clang_C_FLAGS_RELEASE -O2 -g -std=c99 -DNDEBUG)
-set(Clang_C_FLAGS_RELWITHDEBINFO -O2 -g -std=c99 -DNDEBUG)
+if(DEFINED ENV{CRAYPE_VERSION})
```

Comment:
This is to get the -DCRAY_SDK and -ef flags added. 

Also is -DNDEBUG a BML specific flag?

I am open to suggestions about how to do this differently...

BTW, annoyingly, on crusher the C compilers are identified as Clang and the Fortran as Cray. Not sure why.

Reply:
The spock script probably needs to be modified, I'll check.

It's possible commit #8a7df493 would have broken someone else's build script, if offload was being handled manually instead of using BML_OMP_OFFLOAD.

Reply:
-DNDEBUG is not BML specific. It turns off asserts and is typically used in optimized builds

Reply:
you could do something like:
set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O0 -DCRAY_SDK -ef")
to append extra flags to whatever cmake picks by default

Reply:
Yes, it would simplify things if those definitions can be taken out.

Would the same need to be applied to all compilers? We have similar definitions for Intel, XL, etc...I just followed the pattern (some Clang language definitions were already there). 

Maybe redoing this logic should be in a separate PR, it's not really specific to building offload code.

Reply:
Yes, this is an issue that goes beyond that PR. But please try not to add complexity in this particular PR...

Reply:
Sorry, I can't see the complexity I added, the only line mentioned above is the check for CrayPE...

Maybe if you list the line numbers you'd like changed and indicate the specific changes I can try to fix it?

Reply:
Lines 79-100 in CMakeList.txt

Reply:
OK. How are lines 79-86 now?